src/solver.f90

```diff
@@ -203,13 +204,27 @@ subroutine allocate_tdsops(dirps, backend, mesh, der1st_scheme, &
     character(*), intent(in) :: der1st_scheme, der2nd_scheme, &
                                 interpl_scheme, stagder_scheme
 
-    integer :: dir, bc_start, bc_end, n_vert, n_cell, i
+    integer :: dir, bc_start, bc_end, bc_mp_start, bc_mp_end, n_vert, n_cell, i
     real(dp) :: d
 
     dir = dirps%dir
+    d = mesh%geo%d(dir)
+
     bc_start = mesh%grid%BCs(dir, 1)
     bc_end = mesh%grid%BCs(dir, 2)
-    d = mesh%geo%d(dir)
+
+    ! BCs for the midpoint operators, enforce neumann only if the vertex BC
```

Comment:
I would provide a little bit more info here, to explain that because of the FFT, the BC for the pressure has to be NEUMANN. In practice it means that we impose a BC that is not compatible with Navier-Stokes, but that it does not affect the quality of the solution

Reply:
This is a good point, so maybe its not necessary for the iterative Poisson solver. Then this part might need to be bypassed if iterative solver is used instead of FFT.

Added a more detailed description as a comment.